nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/SetSystemPropertiesAction.java

```diff
@@ -45,15 +56,34 @@ protected Computation createComputation(int batchSize, int batchThresholdMs) {
 
     public static class SetSystemPropertyComputation extends AbstractBulkComputation {
 
+        public static final String NOTIFY = "param-notify";
+
         public SetSystemPropertyComputation(String name, int batchSize, int batchThresholdMs) {
             super(name, 1, 1, batchSize, batchThresholdMs);
         }
 
         @Override
         protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
-            ids.forEach(id -> properties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
+            Map<String, Serializable> actualProperties = new HashMap<>(properties);
+            String eventId = (String) actualProperties.remove(NOTIFY);
+            Collection<DocumentRef> refs = ids.stream().map(IdRef::new).collect(Collectors.toList());
+            ids.forEach(id -> actualProperties.forEach((k, v) -> session.setDocumentSystemProp(new IdRef(id), k, v)));
             session.save();
+            if (eventId != null) {
+                EventService eventService = Framework.getService(EventService.class);
+                DocumentModelList docs = session.getDocuments(refs.toArray(new DocumentRef[refs.size()]));
+                docs.forEach(d -> {
+                    DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), d);
```

Comment:
You could extract this to a `raiseEvent` method.

Reply:
do you believe I should ? I'm not really kind of only-called-once methods

Reply:
it will reduce the cognitive complexity, but up to you.

Reply:
I agree, simply returns when eventId is null and extract a method will make thing readable.
Small methods are always good: they express intention and make profiling dead simple when using flame graph.